include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -30,65 +29,72 @@ class NBL_API IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescrip
 		using base_t = asset::IDescriptorSet<const IGPUDescriptorSetLayout>;
 
 	public:
-		IGPUDescriptorSet(core::smart_refctd_ptr<const ILogicalDevice>&& dev, core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout, core::smart_refctd_ptr<IDescriptorPool>&& pool)
-			: base_t(std::move(_layout)), IBackendObject(std::move(dev)), m_version(0ull), m_pool(std::move(pool))
-		{
-            allocateDescriptors();
-
-            for (auto i = 0u; i < asset::EDT_COUNT; ++i)
-            {
-                // There is no descriptor of such type in the set.
-                if (m_descriptorStorageOffsets.data[i] == ~0u)
-                    continue;
+        IGPUDescriptorSet(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& _layout, core::smart_refctd_ptr<IDescriptorPool>&& pool, IDescriptorPool::SDescriptorOffsets&& offsets);
```

Comment:
why is the constructor still public?

Reply:
No idea, probably because of GL. `protected` it now. Resolved?

Reply:
yes